### ISSUE TITLE
Legacy league client ends 4/24, lolnotes will be unusable forever after

### ISSUE THREAD

**Issue by th2pun1sh3r** [bladecoding/LoLNotes#70]
unless an update can be made..

**Comment by th2pun1sh3r** [comment]
http://na.leagueoflegends.com/en/news/game-updates/features/legacy-client-officially-retires-april-24

**Comment by bladecoding** [comment]
Been a bit busy. I'll look into the new client. Assuming they kept the same backend then an update should hopefully be simple. If they move away from the RTMP backend than an update is going take a lot longer. I didn't realize they were completely moving away from adobe air.

**Comment by th2pun1sh3r** [comment]
yeah, lots of new icons and other stuff too - and we need the text note length mod

**Comment by bladecoding** [comment]
Looks like they updated the backend(CEF frontend, WebSocket communication with the backend). I'll keep this ticket updated as I investigate some more. I'll be a bit busy till Saturday. I'll try to minimally post an update end of day Sunday. Post here(I'm now watching the repo) or email me if you don't hear from me by then.

**Comment by th2pun1sh3r** [comment]
would it be a lot of work to just do it from the ground up?

just using the in lobby names

**Comment by th2pun1sh3r** [comment]
similar to this>

https://na.op.gg/multi/query=


https://gyazo.com/53b0a187225593f977268f66923854d3

**Comment by bladecoding** [comment]
The likeliest approach is to hook the client just for the player names and then use a web api to request their data. That way I don't have to essentially repeat all the original work for the RTMP server but this time for WebSockets. With that in mind I may start from scratch. It's still a bit early to say at this point.

**Comment by th2pun1sh3r** [comment]
the only worry I have with the hook approach is league client is getting
increasingly good at detecting 3rd party addons/ scripts/ etc and they are
doing bans over this. Same thing for botted accounts ( I actually
completely stopped selling league smurfs because of that haha )

but I still play the game and notes is really good I feel vulnerable and
blind without it (like some 9 loss ina row player on my team and I don't
know about it)


Also, I think op.gg is way better than lolking these days for querying the
data and it would be awesome to get a brand new program that could show
graphs, main champs etc.

**Comment by bladecoding** [comment]
Are you just looking for an app to paste names into instead? One that provides more information, graphs and lets you rate/note players?

Also you might still be able to read the player names without a hook which would be mostly undetectable(i.e. just reading process memory). I can investigate a read memory only approach instead of a hook.

**Comment by th2pun1sh3r** [comment]
the beauty of notes is how automatic it is, no need to copy paste anything
and as soon as you enter game you can see enemy team stats too (very
valuable information for example u see the enemy has 9 wins in a row so you
play safe against them)

**Comment by bladecoding** [comment]
ya, which sadly comes at the cost of there always being that possibility of being detected/banned.

**Comment by someone32** [comment]
Hey bladecoding, they actually kept the old RTMP, even though there might be some differences the rtmp communication remains on port 2099, doing the exact same requests as the old client. Give it a look, I would also love an update!

**Comment by bladecoding** [comment]
@someone32 I did see a blog post explaining the new infrastructure and how it was still using rtmp. However that blog post is quite old and when I did a quick initial assessment I found that they have moved to websockets. There might still be an rtmp endpoint for stat lookups but that wouldn't help with getting your teammates from the lobby.
